includes/payments/functions.php

```diff
@@ -1508,9 +1513,26 @@ function give_get_donation_form_title( $donation_id, $args = [] ) {
 
 	$args = wp_parse_args( $args, $defaults );
 
-	$form_id     = give_get_payment_form_id( $donation_id );
+	$form_id = give_get_payment_form_id( $donation_id );
+    $form_title = give_get_meta( $donation_id, '_give_payment_form_title', true );
+
+    // Check if the donation form is V3 form
+    if (Utils::isV3Form($form_id)) {
+        $currency = give_get_option('currency');
+        $options = give()->form_meta->get_meta($form_id, '_give_donation_levels', true) ?? [];
+        $donation = Donation::find($donation_id);
+
+        foreach ( $options as $option ) {
+            if (isset($option['_give_amount'], $option['_give_text'])) {
+                if (Money::of($option['_give_amount'], $currency )->getMinorAmount() == $donation->amount->getAmount()) {
+                    return sprintf('%s %s %s', $form_title, $args['separator'], $option['_give_text']);
```

Comment:
Should we be returning early here or overwriting the variables used by the filter?

Reply:
@jonwaldstein This is specific for v3 forms, and the data structure differs from v2 forms. For example, there is no `_give_payment_price_id` in the v3 form, so what happens after this code is important only for v2 forms. I added the filter.

Reply:
Resolved https://github.com/impress-org/givewp/pull/7602/commits/1b41d55072858da78082fc6687366bbc4b6684a6

Reply:
sounds good, thanks!